lib/matplotlib/axes/_axes.py

```diff
@@ -2598,13 +2598,25 @@ def bar_label(self, container, labels=None, *, fmt="%g", label_type="edge",
 
         **kwargs
             Any remaining keyword arguments are passed through to
-            `.Axes.annotate`.
+            `.Axes.annotate`. The alignment parameters (
+            *horizontalalignment* / *ha*, *verticalalignment* / *va*) are
+            not supported because the labels are automatically aligned to
+            the bars.
 
         Returns
         -------
         list of `.Text`
             A list of `.Text` instances for the labels.
         """
+        for key in ['horizontalalignment', 'ha', 'verticalalignment', 'va']:
+            if key in kwargs:
```

Comment:
Might alternatively written as
```suggestion
        unsupported = set(['horizontalalignment', 'ha', 'verticalalignment', 'va']):
        for key in unsupported.intersection(kwargs):
```
But no strong preference. I'll leave this up to you.

Reply:
That should be:
```
unsupported = {'horizontalalignment', 'ha', 'verticalalignment', 'va'}
```

Reply:
Merged as is since the existing code style is good enough.